react/src/components/ImageInstallModal.tsx

```diff
@@ -0,0 +1,197 @@
+      let isGPURequired = false;
+      const imageResource = {} as { [key: string]: any };
+      image?.resource_limits?.forEach((resourceLimit) => {
+        const key = resourceLimit?.key as string;
+        imageResource[key] = resourceLimit?.min;
+      });
+      if ('cuda.device' in imageResource && ' cuda.shares' in imageResource) {
+        isGPURequired = true;
+        imageResource.gpu = 0;
+        imageResource.fgpu = imageResource['cuda.shares'];
+      } else if ('cuda.device' in imageResource) {
+        imageResource.gpu = imageResource['cuda.device'];
+        isGPURequired = true;
+      } else {
+        isGPURequired = false;
+      }
+
+      // Add 256m to run the image.
+      if (imageResource.mem.endsWith('g')) {
+        imageResource.mem = imageResource.mem.replace('g', '.5g');
+      } else if (imageResource.mem.endsWith('m')) {
+        imageResource.mem = Number(imageResource.mem.slice(0, -1)) + 256 + 'm';
+      }
+
+      imageResource.domain = baiClient._config.domainName;
+      imageResource['group_name'] = baiClient.current_group;
+
+      const resourceSlots = await baiClient.get_resource_slots();
+
+      if (isGPURequired) {
+        if (
+          !('cuda.device' in resourceSlots) &&
+          !('cuda.shares' in resourceSlots)
+        ) {
+          delete imageResource['gpu'];
+          delete imageResource['fgpu'];
+          delete imageResource['cuda.shares'];
+          delete imageResource['cuda.device'];
+        }
+      }
+
+      if ('cuda.device' in resourceSlots && 'cuda.shares' in resourceSlots) {
+        // Can be possible after 20.03
+        if ('fgpu' in imageResource && 'gpu' in imageResource) {
+          delete imageResource['gpu'];
+          delete imageResource['cuda.device'];
+        }
+      } else if ('cuda.device' in resourceSlots) {
+        delete imageResource['fgpu'];
+        delete imageResource['cuda.shares'];
+      } else if ('cuda.shares' in resourceSlots) {
+        delete imageResource['gpu'];
+        delete imageResource['cuda.device'];
+      }
+
+      imageResource.enqueueOnly = true;
+      imageResource.type = 'batch';
+      imageResource.startupCommand = 'echo "Image is installed"';
+
+      upsertNotification({
+        message: `${t('environment.InstallingImage')}${imageName}${t('environment.TakesTime')}`,
+        open: true,
+        duration: 2,
+      });
+
+      //@ts-ignore
+      const indicator = await globalThis.lablupIndicator.start('indeterminate');
+      indicator.set(10, t('import.Downloading'));
+      baiClient.image
+        .install(imageName, image?.architecture, imageResource)
+        .then(() => {
+          indicator.end(1000);
+        })
+        .catch((error: any) => {
+          setInstallingImages(
+            imagesToInstall
+              .map((item) => item?.id)
+              .filter(
+                (id): id is string =>
+                  id !== null && id !== undefined && id !== image?.id,
```

Comment:
[ref](https://github.com/lablup/backend.ai-webui/pull/2615/files#diff-8ee05aa0711da52a475b27ba894a607046f46a5c4cf64903702679637cefbef4R141-R148) Isn't this duplicated?